packaging/os/yum.py

```diff
@@ -909,14 +910,16 @@ def latest(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
         if len(pkgs['install']) > 0:    # install missing
             cmd = yum_basecmd + ['install'] + pkgs['install']
             rc, out, err = module.run_command(cmd)
-            res['changed'] = True
+            if not out.strip().lower().endswith("no packages marked for update"):
+                res['changed'] = True
```

Comment:
When screen scraping output we have to make sure that the program returns strings in the language we expect.  Some modules set the locale in one place to do that.  The yum module (probably because it calls several different programs and doesn't screenscrape all of them) sets the environment on each external call.  You can do that with something like the following:
``` python
         lang_env = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C')
         rc, out, err = module.run_command(cmd, environ_update=lang_env)
         if not out.strip().lower().endswith("no packages marked for update"):
             res['changed'] = True
```

And the same for the other place we screenscrape.